### ISSUE TITLE
Bug when trying to open zarr files

### ISSUE THREAD

**Issue by cmdupuis3** [pangeo-forge/pangeo-forge-recipes#454]
@cisaacstern @rabernat 

Charles and I found a bug in trying to write the recipe in pangeo-forge/staged-recipes#239. Running the recipe throws an error, which he realized would also be triggered by running this:

```
import xarray as xr
import fsspec

file = xr.open_dataset(fsspec.open("https://g-402b74.00888.8540.data.globus.org/eddy/low_res.zarr"), engine="zarr")
```

which appears to be the current behavior, where the expected behavior would be:

```
file = xr.open_dataset("https://g-402b74.00888.8540.data.globus.org/eddy/low_res.zarr", engine="zarr")
```

The error I get is `ValueError: Starting with Zarr 2.11.0, stores must be subclasses of BaseStore, if your store exposes the MutableMapping interface wrap it in Zarr.storage.KVStore. Got <OpenFile 'https://g-402b74.00888.8540.data.globus.org/eddy/low_res.zarr'>`

**Comment by rabernat** [comment]
Thanks for opening the issue Chris! I agree this is a bug. We should be able to open zarr files based on their path. This already works in the `beam-refactor` branch. The question is whether it makes sense to back-port a fix to the main branch.

To answer that, I'd like to get a sense from @cisaacstern of **how difficult it would be to deploy a new version of pangeo-forge-recipes to production**? If it is very hard to deploy a new version, then it doesn't make sense implementing a fix, because we won't be able to deploy it anyway. If it is easy, then we can do the fix, release a new version, deploy to prod, and get the recipe shipped.

**Comment by cisaacstern** [comment]
> how difficult it would be to deploy a new version of pangeo-forge-recipes to production?

For a non-breaking change to `pangeo-forge-recipes`, this is easy. It would just require specifying the desired version in [this requirements.txt](https://github.com/pangeo-forge/pangeo-forge-orchestrator/blob/fb5caaf6994d1b73004b447ab00e5687ab941f2e/requirements.txt#L18). (`pangeo-forge-recipes` isn't defined explicitly there at the moment, because it's brought in by `pangeo-forge-runner`.)

**Comment by rabernat** [comment]
Ok, so in that case I will sketch out how this fix looks. Basically, you want Zarr URLs to be treated the same way opendata URLs are currently treated: no caching, and no opening with fsspec...instead they should be passed directly through to xarray.open_dataset.

### Step 1: Add a Zarr filetype here

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/fe5d1f7241e724fde76f86ce582fb0b830af3036/pangeo_forge_recipes/patterns.py#L111-L117

### Step 2: Add cases for Zarr in `pangeo_forge_recipes/recipes/xarray_zarr.py`

Basically just search for every instance of `FileType.opendap` and replace it with `FileType.opendap` OR `FileType.zarr`. Something like

```python
non_openable_types = (FileType.opendap, FileType.zarr)
if config.file_pattern.file_type in non_openable_types:
    # special case for Zarr and opendap
```

### Step 3: Verify that it works

Normally this is where we would add tests. And indeed in the beam-refactor, we cover Zarr inputs extensively, e.g.

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/0e16f15155db31dd13e448c6620e872b158be4b1/tests/conftest.py#L233-L237

However, implementing these tests in the main branch would be a major refactor. The test suite is very confusing for a new contributor. Since we are planning to deprecate this code anyway, I am fine with the idea of taking on technical debt by not testing for Zarr.

But we still need some verification that the feature actually works. You could do this by trying to run the recipe from https://github.com/pangeo-forge/staged-recipes/pull/239 locally.

**Comment by cisaacstern** [comment]
@cmdupuis3 would you like to work on this? With Ryan's roadmap above, it should be straightforward, and I'm happy to consult on slack/video if needed.

**Comment by cisaacstern** [comment]
@cmdupuis3, note that a few of the tests are failing on all PRs to `master` right now for as-yet unknown reasons (xref #451). So if you do start working on this and notice failing tests, don't necessarily assume you've broken anything. We'll either have to resolve #451, or depending on the timeline, just move forward accepting that certain known test failures will happen that are not related to this work.

**Comment by cmdupuis3** [comment]
Sounds good, I'll try to get this done by the end of the week.

**Comment by rabernat** [comment]
Great! The [development guide](https://pangeo-forge.readthedocs.io/en/latest/pangeo_forge_recipes/development/development_guide.html) should help you get started.

**Comment by cmdupuis3** [comment]
Do I need to do anything with this bit?

```
if self.open_input_with_kerchunk:
    raise ValueError("Can't generate references on opendap inputs")
```

It seems specific to Opendap, so I'm leaving it alone for now.

**Comment by cisaacstern** [comment]
> It seems specific to Opendap, so I'm leaving it alone for now.

I agree

**Comment by cisaacstern** [comment]
Closed by #462
